ts/components/dialog/conversationSettings/ConversationTitle.tsx

```diff
@@ -0,0 +1,77 @@
+import { useDispatch } from 'react-redux';
+import { useCurrentUserHasPro } from '../../../hooks/useHasPro';
+import {
+  useNicknameOrProfileNameOrShortenedPubkey,
+  useIsPublic,
+  useIsClosedGroup,
+  useIsMe,
+} from '../../../hooks/useParamSelector';
+import { tr } from '../../../localization/localeTools';
+import type { WithConvoId } from '../../../session/types/with';
+import { H5 } from '../../basic/Heading';
+import { ProIconButton } from '../../buttons/ProButton';
+import { useChangeNickname } from '../../menuAndSettingsHooks/useChangeNickname';
+import { useShowUpdateGroupNameDescriptionCb } from '../../menuAndSettingsHooks/useShowUpdateGroupNameDescription';
+import { showSessionProInfoDialog, SessionProInfoVariant } from '../SessionProInfoModal';
+
+/**
+ * Return the callback to use for the title click event, if one is allowed
+ */
+function useOnTitleClickCb(conversationId: string, editable: boolean) {
+  const changeNicknameCb = useChangeNickname(conversationId);
+  const updateNameDescCb = useShowUpdateGroupNameDescriptionCb({ conversationId });
+  if (!editable) {
+    return null;
+  }
+  return changeNicknameCb || updateNameDescCb;
+}
+
+export const ConversationTitle = ({
+  conversationId,
+  editable,
+}: WithConvoId & { editable: boolean }) => {
+  const dispatch = useDispatch();
+  const nicknameOrDisplayName = useNicknameOrProfileNameOrShortenedPubkey(conversationId);
+  const isCommunity = useIsPublic(conversationId);
+  const isClosedGroup = useIsClosedGroup(conversationId);
+  const isMe = useIsMe(conversationId);
+  const weArePro = useCurrentUserHasPro();
+
+  const onClickCb = useOnTitleClickCb(conversationId, editable);
+
+  // the data-test-id depends on the type of conversation
+  const dataTestId = isCommunity
+    ? 'community-name'
+    : isClosedGroup
+      ? 'group-name'
+      : // for 1o1, this will hold the nickname if set, or the display name
+        'preferred-display-name';
+
+  function onProBadgeClick() {
+    if (weArePro) {
+      return;
+    }
+    showSessionProInfoDialog(SessionProInfoVariant.MESSAGE_CHARACTER_LIMIT, dispatch);
```

Comment:
Seems this is supposed to be a different CTA for when in the profile dialog vs the message char limit CTA, or is this set elsewhere?